diagnostic.md

```diff
@@ -6,70 +6,106 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+To get the books/4 you'd need to do the following:
+1. Change router to say SHOW in routes.rb
+    resources :books, only: [:show]
+2. ?
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+Found in routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+bundle exec is a command to execute a script in the context of the current bundle . 
+
+So bundle exec rake db:migrate executes the rake script with the command db:migrate in the context of the current bundle.
+
+In addtion, Active Record tracks which migrations have already been run so all you have to do is update your source and run rake db:migrate. Active Record will work out which migrations should be run.
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+Rails scaffolding is a quick way to generate some of the major pieces of an application. With the scaffold action, Rails generates all the code it needs dynamically.
+
+The rails g controller things  would generate a route for get '/things' 
+
```

Comment:
The first command generates routes, a controller and model for foo. The second generates just a foo controller.